src/adams_methods.jl

```diff
@@ -0,0 +1,626 @@
+
+                yerr_kp2 = dt * γ_star[steporder + 3] * ϕ[3,steporder+3]
+                err_kp2 = stepsize_hw92!(dt, tdir, y[n], y_np1, yerr_kp2, steporder,
+                                    timeout_VS, dof, abstol, reltol, maxstep, norm)[1]
+
+                # Increment steporder if LE_{k+2} < LE_{k+1} and max(LE_{k},
+                # LE_{k-1}) > LE_{k+1}
+                if err_kp2 < err_kp1
+                    steporder = min(max_order, steporder + 1)
+                    timeout_VO = timeout_VO_const # reset timout_VO
+                end
+            end
+            # Keep timeout_VO 0, or decrement
+            timeout_VO= timeout_VO==0 ? 0 : timeout_VO-1
+        end # end of adaptive order selection
+
+        #--#####################################################################
+        # Prepare for next step
+        #--#####################################################################
+
+        # Shift data from this step, from the "current step" position to the
+        # "previous step" position for ϕstar and ϕ
+        # Note: ϕ[2, :] and ϕ_star[2, :] corresponds to ϕ_n and ϕ*_n, and
+        # ϕ_star[1, :] to ϕ*_{n-1}, etc
+        # TODO: implement with DataStructures.CircularBuffer
+        ϕstar[1, 1:min(max(n-1, 1), current_steporder+1)] =
+                                ϕstar[2, 1:min(max(n-1, 1), current_steporder+1)]
+
+        # Update stepsize variables
+        prev_dt = dt
+        dt = next_dt
+        # Increment our step
+        n = n+1
+    end #while loop over steps
+
+    if points == :steps
+        return vcat(t), y
+    else
+        return vcat(tout), yout
+    end
+end
+
+###############################################################################
+#
+# HELPER FUNCTIONS FOR ADAMS VSVO SOLVER
+#
+###############################################################################
+
+###############################################################################
+# Gamma star function from Hairer et al pg 359, table 1.2
+# used in adaptive step selection (See Hairer pg 423)
+###############################################################################
+#precompute factorial_values
+"compute γ*(j) for higher order versions of ode113"
+function gamma_star_constr(index)
+    γ_star = zeros(Float64, index)
+    #γ_star[1] = 1
+    for j = 1:index
+        γ_star[j] = Polynomials.polyval((1 / factorial(j)) *
+            Polynomials.polyint(Polynomials.poly(diagm(collect(1-(j-1):1)))), 1)
+    end
+    return γ_star
+end
+################################################################################
+# Helper functions for calculating coefficients from recurence relationships
+################################################################################
+#Based on Hairer Lemma 5.1 on page 399
+#Calculates c_j,q and g_n,j for n and j =1:steporder
+"compute c_{j,q} and g_{n,j} coefficients for ode113"
+function g_coefs!(g,c,dt,t,t_np1,n,steporder)
```

Comment:
the argument `steporder` is always called with `steporder+1` or `+2`.  So is this really the steporder?